template/src/ontology/run.sh.jinja2

```diff
@@ -18,6 +18,17 @@ if [ -f run.sh.conf ]; then
     source run.sh.conf
 fi
 
+# Look for a GitHub token
+if [ -n "$GH_TOKEN" ]; then
+    :
+elif [ -f ../../.github/token ]; then
+    GH_TOKEN=$(cat ../../.github/token)
```

Comment:
What if our naive user does not _want_ to share their token with ODK (a potential security risk, from their perspective)?

Reply:
You mean in the case they do have a `GH_TOKEN` variable in their environment, but don’t want to have it passed to the container?

Easy: they can just add

```
GH_TOKEN=invalid
```

to their `run.sh.conf` file. That file is read _before_ we lookup for `GH_TOKEN`, so the bogus definition in `run.sh.conf` will override what they have in their environment. (They can use whatever they want instead of “invalid”; all that matters is that it should not be an empty string.)

Reply:
But it would require the user to know that ODK going to read this variable and do insecure stuff with it. I was more concerned about the 3rd and 4th `elif`, checking default directories for tokens. This seems like something a user would want more control over..

Reply:
Well, they do have control over it: the only way their GitHub token is going to end up in a file called `$XDG_CONFIG_HOME/ontology-development-kit/github/token` is if they deliberately create such a file and write their token into it – it’s not going to happen behind their back. Presumably if they do put their token in such a file, it’s because they do want to have it used by the ODK, because that would be the only purpose of that file.

I would agree with your concerns if we tried to read the token from a non-ODK-specific location (e.g. `$XDG_CONFIG_HOME/github/token`) – users would not legitimately expect that. But they can’t possibly be surprised that the ODK is using the information contained in a file specifically intended for the ODK.

Reply:
Oh yes,  you are of course right. Ok I think I am convinced this is ok then!